src/foundry/common/grid/base.d.mts

```diff
@@ -0,0 +1,614 @@
+   * Always round up to a whole pixel so the pixel is within the grid space (from top-left).
+   * @param x - The x-coordinate grid position
+   * @param y - The y-coordinate grid position
+   * @returns An array representing the position in pixels
+   * @deprecated Since v12 until v14. Use {@link BaseGrid#getTopLeftPoint} instead.
+   */
+  getPixelsFromGridPosition(x: number, y: number): PointArray;
+
+  /* -------------------------------------------- */
+
+  /**
+   * Shift a pixel position [x,y] by some number of grid units dx and dy
+   * @param x  - The starting x-coordinate in pixels
+   * @param y  - The starting y-coordinate in pixels
+   * @param dx - The number of grid positions to shift horizontally
+   * @param dy - The number of grid positions to shift vertically
+   * @param options - Additional options to configure shift behavior.
+   * @deprecated Since v12 until v14. Use {@link BaseGrid#getShiftedPoint} instead.
+   */
+  shiftPosition(
+    x: number,
+    y: number,
+    dx: number,
+    dy: number,
+    options?: InexactPartial<{
+      /**
+       * The token that is being shifted.
+       */
+      token?: Token;
+    }>,
+  ): PointArray;
+
+  /* -------------------------------------------- */
+
+  /**
+   * Measure the distance traversed over an array of measured segments
+   * @param segments - An Array of measured movement segments
+   * @param options  - Additional options which modify the measurement
+   *                   (default: `{}`)
+   * @returns An Array of distance measurements for each segment
+   * @deprecated Since v12 until v14. Use {@link BaseGrid#measurePath} instead.
+   */
+  measureDistances(segments: GridLayer.Segment[], options?: GridLayer.MeasureDistancesOptions): number[];
+
+  /* -------------------------------------------- */
+
+  /**
+   * Given a pair of coordinates (x1,y1), return the grid coordinates (x2,y2) which represent the snapped position
+   * Under a "gridless" system, every pixel position is a valid snapping position
+   *
+   * @param x        - The exact target location x
+   * @param y        - The exact target location y
+   * @param interval - An interval of grid spaces at which to snap.
+   *                   At interval=1, snapping occurs at pixel intervals defined by the grid size
+   *                   At interval=2, snapping would occur at the center-points of each grid size
+   *                   At interval=null, no snapping occurs
+   *                   (default: `null`)
+   * @param options  - Additional options to configure snapping behavior.
+   * @returns An object containing the coordinates of the snapped location
+   * @deprecated Since v12 until v14. Use {@link BaseGrid#getSnappedPoint} instead.
+   */
+  getSnappedPosition(
+    x: number,
+    y: number,
+    interval?: number | null,
+    options?: InexactPartial<{
+      /** The token that is being moved. */
+      token?: Token;
```

Comment:
General point - don't need to include the optional ? if you're also using InexactPartial